astropy/coordinates/transformations.py

```diff
@@ -70,6 +70,20 @@ def frame_set(self):
 
         return self._cached_frame_set.copy()
 
+    @property
+    def frame_attrnames_set(self):
+        """
+        A `frozenset` of all the attribute names of all frame classes in this `TransformGraph`.
+        """
+        if self._cached_frame_attrnames_set is None:
+            result = set()
+            for frame_cls in self.frame_set:
+                for attr in frame_cls.get_frame_attr_names().keys():
```

Comment:
I'm not quite sure but is `frame_cls.get_frame_attr_names()` a dictionary? If so then `result.update(frame_cls.get_frame_attr_names())` can also avoid re-hashing the "keys". On the other hand `.keys()` creates a list-like-iterator where `set.update` needs to hash the values again.

If it's not a dictionary then: Forget what I was saying above 😄 

But yes, keep it as-is, there is no actual (performance) reason to change it.